cos-fleetshard-sync/src/main/java/org/bf2/cos/fleetshard/sync/resources/ConnectorStatusUpdater.java

```diff
@@ -3,31 +3,69 @@
 import javax.enterprise.context.ApplicationScoped;
 import javax.inject.Inject;
 
+import io.micrometer.core.instrument.Gauge;
+import io.micrometer.core.instrument.MeterRegistry;
+import io.micrometer.core.instrument.Tag;
+import org.bf2.cos.fleet.manager.model.ConnectorDeploymentStatus;
 import org.bf2.cos.fleetshard.api.ManagedConnector;
+import org.bf2.cos.fleetshard.sync.FleetShardSyncConfig;
 import org.bf2.cos.fleetshard.sync.client.FleetManagerClient;
 import org.bf2.cos.fleetshard.sync.client.FleetManagerClientException;
 import org.bf2.cos.fleetshard.sync.client.FleetShardClient;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
+import java.util.List;
+import java.util.concurrent.atomic.AtomicInteger;
+
 @ApplicationScoped
 public class ConnectorStatusUpdater {
     private static final Logger LOGGER = LoggerFactory.getLogger(ConnectorStatusUpdater.class);
 
+    public static final String CONNECTOR_STATE = "connector.state";
+    private List<Tag> tags;
+
+
     @Inject
     FleetManagerClient fleetManagerClient;
     @Inject
     FleetShardClient connectorClient;
+    @Inject
+    MeterRegistry registry;
+    @Inject
+    FleetShardSyncConfig config;
 
     public void update(ManagedConnector connector) {
         LOGGER.debug("Update connector status (name: {}, phase: {})",
             connector.getMetadata().getName(),
             connector.getStatus().getPhase());
 
         try {
-            fleetManagerClient.updateConnectorStatus(
-                connector,
-                ConnectorStatusExtractor.extract(connector));
+            ConnectorDeploymentStatus connectorDeploymentStatus = ConnectorStatusExtractor.extract(connector);
+
+            fleetManagerClient.updateConnectorStatus(connector, connectorDeploymentStatus);
+
+            LOGGER.debug("Updating change status metrics (Connector_id : {}, state: {})",
+                connector.getSpec().getConnectorId(),
+                ConnectorStatusExtractor.extract(connector).getPhase());
+
+            switch (connectorDeploymentStatus.getPhase()) {
+                case READY:
+                    measure(connector, 1);
```

Comment:
don't use magic values like 1,2,3,4,5, constants would be better or alternatively the PhaseType enum can be enhanced to include additional values.

Reply:
Agreed. I'd rather we use `connector.getStatus().getPhase()` for reporting, and augmenting the PhaseType to hold the enum values instead of constants that don't belong to that class.

Reply:
@rinaldodev is there any change related to this you want before merging ?

Reply:
@rinaldodev, Sorry I didn't see this comment the last time. I agree with you that we should introduce int values with PhaseType instead of constants.
My suggestion will be to also re-evaluate both ConnectorState and PhaseType enums and keep only one such type that confirms the current status/state of the connector (if they don't provide a different kind of information about the connector).